src/components/hardware/HardwareWalletTheme.tsx

```diff
@@ -54,19 +56,27 @@ export function ArConnectThemeProvider({ children }: PropsWithChildren<{}>) {
         theme === "dark" ? ARCONNECT_DARK_THEME : ARCONNECT_LIGHT_THEME
       )}
     >
-      <ThemeBackgroundObserver />
+      <ThemeBackgroundObserver theme={theme} />
 
       {children}
     </ThemeProvider>
   );
 }
 
-export function ThemeBackgroundObserver() {
+interface ThemeBackgroundObserverProps {
+  theme?: DisplayTheme;
+}
+
+export function ThemeBackgroundObserver({
+  theme
+}: ThemeBackgroundObserverProps) {
   const styledComponentsTheme = useStyledComponentsTheme();
   const backgroundColor = styledComponentsTheme.background;
   const textColor = styledComponentsTheme.primaryText;
 
   useEffect(() => {
+    if (!theme) return;
```

Comment:
`theme` becomes `undefined` when the extension popup closes, which was setting the wrong value in `localStorage` / CSS vars on the first render.